src/sentry/api/serializers/models/organization.py

```diff
@@ -653,10 +657,30 @@ def serialize(  # type: ignore[explicit-override, override]
             team__organization=obj
         ).count()
 
-        sample_rate = quotas.backend.get_blended_sample_rate(organization_id=obj.id)
-        context["isDynamicallySampled"] = (
-            has_dynamic_sampling(obj) and sample_rate is not None and sample_rate < 1.0
-        )
+        is_dynamically_sampled = False
+        sample_rate = None
+        if has_dynamic_sampling(obj):
+            if (
+                obj.get_option("sentry:sampling_mode", SAMPLING_MODE_DEFAULT)
+                == DynamicSamplingMode.ORGANIZATION
+            ):
+                sample_rate = obj.get_option(
+                    "sentry:target_sample_rate",
+                    quotas.backend.get_blended_sample_rate(organization_id=obj.id),
+                )
+                is_dynamically_sampled = sample_rate is not None and sample_rate < 1.0
```

Comment:
We need to perform a couple of checks, think of it like a waterfall:

- if `dynamic-sampling-custom`:
  - if `Mode.ORG`: use `org.get_option("target_sample_rate", TARGET_SAMPLE_RATE_DEFAULT)`
  - elif `Mode.PROJECT`: check `ProjectOption.filter(..)`
- elif `dynamic-sampling`: use `get_blended_sample_rate`